documentation/src/main/asciidoc/public/reference/_search-dsl-predicate.adoc

```diff
@@ -1504,6 +1504,56 @@ include::{sourcedir}/org/hibernate/search/documentation/search/predicate/Predica
 ----
 ====
 
+[[search-dsl-predicate-knn-with-text-search]]
+=== Combining `knn` with other predicates
+
+A `knn` predicate can, be combined with the regular text-search predicates. It can improve the quality of search results
+by increasing the score of documents that are more relevant based on vector embeddings characteristics:
+
+.Enriching regular text search with K-Nearest Neighbors search
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java[tags=knn-and-match]
+----
+====
+
+[[search-dsl-predicate-knn-limitations]]
+=== Backend specifics and limitations
+
+With the Elasticsearch backend a `knn` predicate can only be added as a top-level predicate,
+i.e. a predicate directly passed to a where clause of a search query,
+or as part of a top-level disjunction,
+i.e. as `should` clauses of a top-level <<search-dsl-predicate-boolean,`boolean` predicate>>
+(the same can be achieved by using an <<search-dsl-predicate-or,`or` predicate>>).
+Any other usages of a `knn` predicate, with this backend, would lead to an exception being thrown.
```

Comment:
"Elasticsearch backend" can technically be either Elasticsearch or OpenSearch. Try this:

```suggestion
With the Elasticsearch backend and when using <<backend-elasticsearch-compatibility-elasticsearch,Elasticsearch>>,
 a `knn` predicate can only be added as a top-level predicate,
i.e. a predicate directly passed to a where clause of a search query,
or as part of a top-level disjunction,
i.e. as `should` clauses of a top-level <<search-dsl-predicate-boolean,`boolean` predicate>>
(the same can be achieved by using an <<search-dsl-predicate-or,`or` predicate>>).
Any other usages of a `knn` predicate, with this backend, would lead to an exception being thrown.

<<backend-elasticsearch-compatibility-opensearch,OpenSearch>> does not have this limitation.
```

I'd also suggest to update the sections I linked to, to list this limitation and add a link back to this section.